src/kili/services/copy_project/__init__.py

```diff
@@ -119,3 +157,154 @@ def _copy_quality_settings(self, new_project_id: str, src_project: Dict) -> None
             use_honeypot=src_project["useHoneyPot"],
             review_coverage=src_project["reviewCoverage"],
         )
+
+    def _copy_assets(self, from_project_id: str, new_project_id: str, batch_size: int = 200):
+        """
+        Copy assets from a project to another.
+
+        Fetches assets by batch since `content` urls expire.
+        """
+        nb_assets_to_copy = self.count_assets(project_id=from_project_id)
+        skip = 0
+        with tqdm(total=nb_assets_to_copy) as pbar:
+            while skip < nb_assets_to_copy:
+                assets = self.assets(
+                    project_id=from_project_id,
+                    fields=["content", "externalId"],
+                    skip=skip,
+                    first=batch_size,
+                    disable_tqdm=True,
+                )
+                self.append_many_to_dataset(
+                    project_id=new_project_id,
+                    content_array=[x["content"] for x in assets],
+                    external_id_array=[x["externalId"] for x in assets],
+                    disable_tqdm=True,
+                )
+                skip += len(assets)
+                pbar.update(skip)
+
+    # pylint: disable=too-many-locals
+    def _copy_labels(self, from_project_id: str, new_project_id: str):
+        assets_new_project = self.assets(
+            new_project_id, fields=["externalId", "id"], disable_tqdm=True
+        )
+        assets_new_project = {asset["externalId"]: asset["id"] for asset in assets_new_project}
```

Comment:
this is not the same object, the first one is a list and this one is a dict. You can rather call it  `assets_new_project_id_map`